src/cGame_logic.cpp

```diff
@@ -573,34 +569,7 @@ void cGame::loadSkirmishMaps() const
 
 void cGame::shakeScreenAndBlitBuffer()
 {
-    if (m_TIMER_shake == 0) {
-        m_TIMER_shake = -1;
-    }
-
-    // only in playing state we shake screen
-    if (m_state == GAME_PLAYING) {
-        // TODO: move the shaking part of the rendering in the playing state object at some time
-        // and shake it within the 'bmp_screen', so that the actual double buffering (bmp_screen -> screen) happens
-        // always at some point in the main loop, and does not need to know about the shaking logic.
-
-        // blitSprite on screen
-        if (m_TIMER_shake > 0) {
-            // the more we get to the 'end' the less we 'throttle'.
-            // Structure explosions are 6 time units per cell.
-            // Max is 9 cells (9*6=54)
-            // the max border is then 9. So, we do time / 6
-            int shakiness = std::min(m_TIMER_shake, 69);
-            float offset = mapCamera->factorZoomLevel(std::min(shakiness / 5, 9));
-
-            m_shakeX = -abs(offset / 2) + RNG::rnd(offset);
-            m_shakeY = -abs(offset / 2) + RNG::rnd(offset);
-
-            // @Mira recreate shake screen
-        }
-        else {
-            fadeOutOrBlitScreenBuffer();
-        }
-    }
+    m_screenShake->update(m_state, GAME_PLAYING, mapCamera);
```

Comment:
since we only want to shake screen for `GAME_PLAYING` I would suggest we run this update logic within the `game playing` update loop.  Then we can get rid of the `m_state/GAME_PLAYING` mechanism and remove the 'knowledge of state' in the `m_screenShake` (`cScreenShake`) class altogether. Then it will be agnostic and we can let the screen shake whenever we want eventually.